darts/ad/anomaly_model.py

```diff
@@ -0,0 +1,1003 @@
+        future_covariates
+            An optional future-known covariate series. This applies only if the model supports future covariates.
+        forecast_horizon
+            The forecast horizon for the predictions.
+        start
+            The first point of time at which a prediction is computed for a future time.
+            This parameter supports 3 different data types: ``float``, ``int`` and ``pandas.Timestamp``.
+            In the case of ``float``, the parameter will be treated as the proportion of the time series
+            that should lie before the first prediction point.
+            In the case of ``int``, the parameter will be treated as an integer index to the time index of
+            `series` that will be used as first prediction time.
+            In case of ``pandas.Timestamp``, this time stamp will be used to determine the first prediction time
+            directly.
+        num_samples
+            Number of times a prediction is sampled from a probabilistic model. Should be left set to 1 for
+            deterministic models.
+        return_model_prediction
+            Boolean value indicating if the prediction of the model should be returned along the anomaly score
+            Default: False
+
+        Returns
+        -------
+        Union[TimeSeries, Sequence[TimeSeries]]
+            Anomaly score time series generated by a scorer (list of anomaly score if more than one scorer)
+        """
+
+        raise_if_not(
+            self.model._fit_called,
+            f"Model {self.model} has not been trained. Please call .fit()",
+        )
+
+        list_series = [series] if not isinstance(series, Sequence) else series
+
+        list_past_covariates = self._check_covariates(
+            past_covariates, list_series, "past"
+        )
+        list_future_covariates = self._check_covariates(
+            future_covariates, list_series, "future"
+        )
+
+        self._check_window_size(list_series, start)
+
+        list_pred = []
+        for idx, s in enumerate(list_series):
+
+            if list_past_covariates is not None:
+                past_covariates = list_past_covariates[idx]
+
+            if list_future_covariates is not None:
+                future_covariates = list_future_covariates[idx]
+
+            list_pred.append(
+                self._predict_with_forecasting(
+                    s,
+                    past_covariates=past_covariates,
+                    future_covariates=future_covariates,
+                    forecast_horizon=forecast_horizon,
+                    start=start,
+                    num_samples=num_samples,
+                )
+            )
+
+        list_anomaly_scores = []
+        for scorer in self.scorers:
+            list_anomaly_scores.append(
+                scorer.score(
+                    list_pred if len(list_pred) > 1 else list_pred[0],
```

Comment:
I recently added this condition, to respect a rule: the return type of a function must match the given input type. 

The scorer does accept series OR a list of series. And the rule is:
-  if a series is given -> will return an anomaly series  
- if a list of series is given -> will return a list of anomaly series (even if the series contains only 1 element)
I wanted that the function returns the same type of input. 

The situation here is that our anomaly_model accepts series OR a list of series. And if a series is given, it will be converted into a list, which means that anomaly_model will always manipulate list. As a consequence, when the anomaly_model calls the scorers, it will give as input a list for list_pred and list_series.

However, if the users input was a series and not a list for the anomaly_model, the scorer will nevertheless receive a list, and will return a list of anomaly scores rather than a series of anomaly scores. 

To avoid that, i added these conditions here...
I hope it's clear! And hopefully, there is a cleaner way to solve this issue.

Reply:
Couldn't you do something like
```python
for scorer in self.scorers:
    list_anomaly_scores += scorer.score(list_pred, list_series)
```
This way all scorers always return a list (could be of length 1), and these lists get concatenated into `list_anomaly_scores`.